### ISSUE TITLE
Image upload empty json response

### ISSUE THREAD

**Issue by juanmc2005** [gotev/android-upload-service#71]
I'm trying to upload an image and the server is supposed to always respond a 200 OK with a json payload. I tested it using postman and it worked ok, but when i'm using the library the response body arrives empty sometimes.
The code i'm using is the following:

public void uploadProfileImage(String absolutePath, String uuid, String accessToken,
```
ImageUploadListener listener) {
      try {
            new MultipartUploadRequest(mContext, uuid, UrlConstants.UPLOAD_PROFILE_PHOTO)
                    .addFileToUpload(absolutePath, "profileImage")
                    .addHeader("X-AUTHORITY", accessToken)
                    .setMaxRetries(2)
                    .startUpload();
            listener.onUploadRequestSent();
        } catch (Exception exc) {
            listener.onUploadError();
        }
}
```
Any ideas?

Thanks

**Comment by alexbbb** [comment]
Please show me the code with which you get the response from the server. The code to create the request is ok.

**Comment by juanmc2005** [comment]
```
public MainPresenter(MainActivity activity) {
        mActivity = activity;
        mUploadServiceReceiver = new UploadServiceBroadcastReceiver() {
            @Override
            public void onError(String uploadId, Exception exception) {
                mActivity.displayError(mActivity.getString(R.string.image_upload_error));
            }

            @Override
            public void onCompleted(String uploadId, int serverResponseCode, String serverResponseMessage) {
                // Here, serverResponseCode is 200, serverResponseMessage is ""
                Log.d("JSON", serverResponseMessage);
                mActivity.displayToast(mActivity.getString(R.string.updated_profile));
            }

            @Override
            public void onProgress(String uploadId, int progress) {
                Log.d("PROGRESS", String.valueOf(progress));
            }
        };
}
```

**Comment by alexbbb** [comment]
Can't reproduce your issue. I've used the [example node.js demo server and the demo android app] (https://github.com/alexbbb/android-upload-service#examples-) and I can correctly see the `serverResponseMessage` string in the `onCompleted` listener. 
I encourage you to checkout the sources and try the demo node.js server yourself with your app. Follow the instructions in the readme and in the command line. By default the server responds with the string `Upload Ok!` in the response body, but you can change that [by editing this line](https://github.com/alexbbb/android-upload-service/blob/master/examples/server-nodejs/index.js#L101) to make your tests. Let me know :)

**Comment by alexbbb** [comment]
@juanmc2005 have you tried the proposed test or resolved your server side issue?

**Comment by juanmc2005** [comment]
Sorry I had to switch to a manual Volley implementation, I couldn't solve it

**Comment by alexbbb** [comment]
Your issue made me revisit the library's code to try to understand why it won't work in your scenario. I've found that currently version 1.6 supports only string based server body responses, terminated with a new line. If for some reason the response is not in UTF-8 or does not contain a new line at the end, you will end up with the empty `serverResponseMessage` string. I've fixed that in #76 and it will be available in the upcoming 2.0 release. Just wanted to let you know about that.
